### PR TITLE
Fix issues while deploying management packages into bootstrap and management cluster

### PR DESCRIPTION
**Pull request by anujc25**
### What this PR does / why we need it

As part of #1668, we added support for deploying management packages into bootstrap and management cluster. However, there are few gaps that are discovered when running end to end functionality. This PR tries to address those issues.

- Skip management repository install using ytt overlays when `package-based-lcm-beta` feature flag is enabled (This is not needed because when this flag is enabled, management repository is installed as part of bootstrapping management cluster programmatically)

- Fix issues around management repository reconcile wait getting stuck after successful installation

…

### Which issue(s) this PR fixes

Related #1664, #1665, #1666

### Describe testing done for PR
- Create management cluster with `package-based-lcm-beta` feature flag enabled
- Verify the `tanzu-management` package repository is installed and `tkg` composite package is installed correctly

### Release note

```release-note
Fix issues while deploying management packages into bootstrap and management cluster
```

### PR Checklist

- [x] Squash the commits into one or a small number of logical commits
      
- [x] Use good commit [messages](https://github.com/vmware-tanzu/tanzu-framework/blob/main/CONTRIBUTING.md)
- [x] Ensure PR contains terms all contributors can understand and links all contributors can access

### CODE REVIEW

File: pkg/v1/providers/ytt/02_addons/management-packages/add_management_package_repo.yaml [hunk 1]
```diff
@@ -17,7 +17,7 @@ spec:
       image: #@ "{}/{}:{}".format(get_image_repo_for_component(management_package_repo_bundle), management_package_repo_bundle.imagePath, management_package_repo_bundle.tag)
 #@ end
 
-#@ if data.values.TKG_CLUSTER_ROLE != "workload" and data.values.PROVIDER_TYPE != "tkg-service-vsphere":
+#@ if data.values.TKG_CLUSTER_ROLE != "workload" and data.values.PROVIDER_TYPE != "tkg-service-vsphere" and not data.values.FEATURE_FLAG_PACKAGE_BASED_LCM:
```
Comment (saji-pivotal) [inline pkg/v1/providers/ytt/02_addons/management-packages/add_management_package_repo.yaml:20]: With ytt overlays, does `not data.values.FEATURE_FLAG_PACKAGE_BASED_LCM` mean `FEATURE_FLAG_PACKAGE_BASED_LCM: false` or can it also mean that the `FEATURE_FLAG_PACKAGE_BASED_LCM` is empty?

Comment (saji-pivotal) [inline pkg/v1/providers/ytt/02_addons/management-packages/add_management_package_repo.yaml:20]: With ytt overlays, does `not data.values.FEATURE_FLAG_PACKAGE_BASED_LCM` mean `FEATURE_FLAG_PACKAGE_BASED_LCM: false` or can it also mean that the `FEATURE_FLAG_PACKAGE_BASED_LCM` is empty?

File: pkg/v1/tkg/managementcomponents/management_component_install.go [hunk 2]
```diff
@@ -100,7 +100,7 @@ func installManagementPackageRepository(pkgClient tkgpackageclient.TKGPackageCli
 	repositoryOptions.PollInterval = packagePollInterval
 	repositoryOptions.PollTimeout = packagePollTimeout
 
-	return pkgClient.UpdateRepositorySync(repositoryOptions, tkgpackagedatamodel.OperationTypeInstall)
+	return pkgClient.UpdateRepositorySync(repositoryOptions, tkgpackagedatamodel.OperationTypeUpdate)
```
Comment (saji-pivotal) [inline pkg/v1/tkg/managementcomponents/management_component_install.go:103]: I'm just curious, why has the operationType been updated here?

Comment (saji-pivotal) [inline pkg/v1/tkg/managementcomponents/management_component_install.go:103]: I'm just curious, why has the operationType been updated here?

File: pkg/v1/providers/ytt/02_addons/management-packages/add_management_package_repo.yaml (hunk 1 above)
Comment (anujc25) [reply pkg/v1/providers/ytt/02_addons/management-packages/add_management_package_repo.yaml:20]: It applies to both. Basically `not data.values.FEATURE_FLAG_PACKAGE_BASED_LCM` mean either it's `false` or `empty` or `None` then overall condition is `true`

File: pkg/v1/tkg/managementcomponents/management_component_install.go (hunk 2 above)
Comment (anujc25) [reply pkg/v1/tkg/managementcomponents/management_component_install.go:103]: It was a bug that got discovered during testing. When updating the repository to properly update and return the correct error on failure this needs to be set to `OperationTypeUpdate` operation stucks in waiting state.

**Review by saji-pivotal — APPROVED** [review]
LGTM from me. It might be helpful to get a few more eyeballs on this

**Review by vuil — APPROVED** [review]
LGTM, thanks!

**Pull request merged**